external/urlfrontier/src/test/java/com/digitalpebble/stormcrawler/urlfrontier/StatusUpdaterBoltTest.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.digitalpebble.stormcrawler.urlfrontier;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.digitalpebble.stormcrawler.Metadata;
+import com.digitalpebble.stormcrawler.TestOutputCollector;
+import com.digitalpebble.stormcrawler.TestUtil;
+import com.digitalpebble.stormcrawler.persistence.Status;
+import java.util.HashMap;
+import java.util.concurrent.*;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.tuple.Tuple;
+import org.junit.*;
+import org.junit.rules.Timeout;
+
+public class StatusUpdaterBoltTest {
+    private StatusUpdaterBolt bolt;
+    private TestOutputCollector output;
+    private URLFrontierContainer urlFrontierContainer;
+
+    private static final String persistedKey = "somePersistedKey";
+    private static final String notPersistedKey = "someNotPersistedKey";
+
+    private static ExecutorService executorService;
+
+    @Rule public Timeout globalTimeout = Timeout.seconds(60);
+
+    @BeforeClass
+    public static void beforeClass() {
+        executorService = Executors.newFixedThreadPool(2);
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        executorService.shutdown();
+        executorService = null;
+    }
+
+    @Before
+    public void before() {
+
```

Comment:
I'd rather not have to change a hardcoded version number in a class when it won't be used most of the time. Let's do this instead - it works fine.

```
String image = "crawlercommons/url-frontier";

String version = System.getProperty("urlfrontier-version");
if (version != null)
  image += ":" + version;

  urlFrontierContainer = new URLFrontierContainer(image);
  urlFrontierContainer.start(); 
```